resources/js/main.js

```diff
@@ -0,0 +1,287 @@
+            return t('plugins.generic.plagiarism.similarity.action.generateReport.confirmation');
+        }
+
+        return t('plugins.generic.plagiarism.similarity.action.refreshReport.confirmation');
+    }
+
+    function getActionUrl(fileStatus)
+    {
+        if (!fileStatus.ithenticateId) {
+            return fileStatus.ithenticateUploadUrl;
+        }
+    
+        if (!fileStatus.ithenticateSimilarityScheduled) {
+            return fileStatus.ithenticateReportScheduleUrl;
+        }
+    
+        return fileStatus.ithenticateReportRefreshUrl;
+    }
+
+    function isEulaConfirmationRequired(contextStatus, submissionStatus, userStatus)
+    {
+        // Check if EULA confirmation required for this tenant
+        if (!contextStatus.eulaRequired) {
+            return false;
+        }
+
+        // If no EULA is stamped with submission
+		// means submission never passed through iThenticate process
+        if (!submissionStatus.ithenticateEulaVersion) {
+            return true;
+        }
+
+        // If no EULA is stamped with submitting user
+		// means user has never previously interacted with iThenticate process
+        if (!userStatus.ithenticateEulaVersion) {
+            return true;
+        }
+
+        // If user and submission EULA do not match
+		// means users previously agreed upon different EULA
+        if (submissionStatus.ithenticateEulaVersion !== userStatus.ithenticateEulaVersion) {
+            return true;
+        }
+
+        return false;
+    }
+
+    async function executePlagiarismAction(fileStatus)
+    {
+        const actionUrl = getActionUrl(fileStatus);
+
+        const { apiUrl } = useUrl(actionUrl);
+    
+        const { 
+            fetch: executeIthenticateAction, 
+            data: ithenticateActionData,
+        } = useFetch(apiUrl);
+        
+        await executeIthenticateAction();
+
+        return ithenticateActionData;
+    }
+
+    fileStore.extender.extendFn('getItemActions', (originalResult, args) => {
+        const submission = fileStore.props.submission;
+        const submissionFile = isOPS() ? args.galley.file : args.file;
+
+        if (!isOPS() && (submission.stageId !== submissionStageId)) {
+            return [...originalResult];
+        }
+
+        if (ithenticateStatus.value) {
+            const fileStatus = deduceFileStatus(submissionFile, ithenticateStatus.value);
+            const userStatus = ithenticateStatus.value?.user;
+            const submissionStatus = ithenticateStatus.value?.submission;
+            const contextStatus = ithenticateStatus.value?.context;
+
+            // Action on non allowed file is restricted
+            if (!fileStatus?.ithenticateUploadAllowed) {
+                return [...originalResult];
+            }
+
+            const matchAllowedRoles = userStatus
+                .ithenticateActionAllowedRoles
+                .filter(role => window.pkp.currentUser.roles.includes(role));
```

Comment:
This is perfectly fine. Just maybe there is opportunity to use the useCurrentUser composable, which has `hasCurrentUserAtLeastOneRole` method.